exc2/src/harmonicFunctions.py

```diff
@@ -0,0 +1,95 @@
+"""
+Author: Gustav Kjellberg
+Email: kjellberg.p.gustav@gmail.com
+Licence: MIT
+Creation data: 22 Sept 2020
+Last modified: 23 Sept 2020
+"""
+
+import math as math
+
+import numpy as np
+
+class HarmonicFunctions():
+    """[summary]
+        A class that creates a harmonic function.
+    """    
+    def __init__(self, t_start, t_stop):
+        """[summary]
+            Class initializer
+        Args:
+            t_start ([int]): [The time t where the data starts]
+            t_stop ([int]): [The time t where the data ends]
+        """        
+        self._t_start = t_start
+        self._t_stop = t_stop
+        self._global_minima_value = math.inf
+        self._step_size = 0.01
+        self._global_minima_t = None
+        self._periodicity = None
+        self._index_of_first_minima = None
+
+    def find_period(self, h_0, h_1, value_t, value_t_minus_1,t):
+        """[summary]
+            Find the period of the function given that we have enough of a span.
+        Args:
+            h_0 ([float]): [function value at t_start]
+            h_1 ([float]): [function value at t_start+1 step size]
+            value_t ([float]): [function value at current t]
+            value_t_minus_1 ([float]): [function value at previous t]
+            t ([float]): [current t]
+        """        
+        if math.isclose(value_t, h_0, abs_tol=0.00001): #rounding errors in python, accepts a small diff.
+            direction_start = False if h_1-h_0 < 0 else True
+            direction_end = False if value_t-value_t_minus_1 < 0 else True
+            if direction_start is direction_end:
```

Comment:
```suggestion
            if direction_start == direction_end:
```

Try not to use `is` unless it's appropriate.